### PR TITLE
feat(accounts): implement birthday prop

### PR DESCRIPTION
**Pull request by st3iny**
* Resolves: # 

## Summary

cc @szaimen @JohannesGGE 

## TODO

- [ ] bug: reloading does not show new and set birthday
- [ ] other todo comments

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/CardDAV/Converter.php [hunk 1]
```diff
@@ -128,6 +128,10 @@ public function createCardFromUser(IUser $user): ?VCard {
 				case IAccountManager::PROPERTY_TWITTER:
 					$vCard->add(new Text($vCard, 'X-SOCIALPROFILE', $property->getValue(), ['TYPE' => 'TWITTER', 'X-NC-SCOPE' => $scope]));
 					break;
+				// TODO
+				case IAccountManager::PROPERTY_BIRTHDAY:
+					$vCard->add(new Text($vCard, 'BIRTHDAY', $property->getValue(), ['TYPE' => 'BIRTHDAY', 'X-NC-SCOPE' => $scope]));
```
Comment (szaimen) [inline apps/dav/lib/CardDAV/Converter.php:133]: todo

File: apps/settings/src/components/PersonalInfo/BirthdaySection.vue [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	},
+
+	methods: {
+		onInput(e) {
+			console.log('onInput', e)
+			this.birtdayValue = e
+			this.debouncePropertyChange(this.birthdayValue.toString())
+		},
+
+		debouncePropertyChange: debounce(async function(value) {
```
Comment (szaimen) [inline apps/settings/src/components/PersonalInfo/BirthdaySection.vue:93]: todo

File: apps/dav/lib/CardDAV/Converter.php (hunk 1 above)
Comment (st3iny) [reply apps/dav/lib/CardDAV/Converter.php:133]: BDAY (https://www.rfc-editor.org/rfc/rfc6350#section-6.2.5)

**Comment by szaimen** [comment]
Unfortunately Richard and me did not succeed in publishing the birthday calendar to the calendar app yet.

There are two ways how this could be accomplished in theory that we thought about:
1. Making the system birthday calendar publicly available by sharing it with each user and rewriting the whole logic around the personal birthday calendar in order to merge it into the system birthday calendar 
2. Making the system birthday calendar available by sharing it with each user and couple it in the frontend to the personal birthday calendar and always hide it from the list - so that if the birthday calendar gets deactived, the system birthday calendar also get disabled hidden which is probably the best solution to the problem.

**Comment by tcitworld** [comment]
2. looks the best one…until you get some user that wants their personal birthday calendar but not the system birthday calendar

Another (more complex) possibility is to have something like https://github.com/nextcloud/calendar/issues/4453. You would manually pick addressbooks you want to consider for the birthday calendar, including the system one.

**Comment by st3iny** [comment]
Yeah, well that would be a great extension. The complexity of the second option is already high enough for my liking. We will try to have another look at the feature if time permits.

**Comment by szaimen** [comment]
I guess this can be closed now that https://github.com/nextcloud/server/pull/45512 was merged?

**Comment by st3iny** [comment]
True. Your and Thomas' comment from earlier are worthy of archiving if anyone ever picks up the birthday calendar feature again.

**Comment by aurisnoctis** [comment]
> True. Your and Thomas' comment from earlier are worthy of archiving if anyone ever picks up the birthday calendar feature again.

Does this mean that https://github.com/nextcloud/server/pull/45512 does not cover integrating the birthdays into a calendar that everyone can see / subscribe to?

**Comment by st3iny** [comment]
Yes, there is currently no system-wide birthday calendar for users. This is very hard to implement correctly and the reasons are stated above.

**Pull request closed without merging**